core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -0,0 +1,150 @@
+    samDAO.userHasAction(SamResourceTypeNames.billingProject, projectName.value, action, userInfo).flatMap {
+      case true => op
+      case false => Future.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Forbidden, s"You cannot perform ${action.value} on project ${projectName.value}")))
+    }
+  }
+
+  def requireAlphaUser[T]()(op: => Future[T]): Future[T] = {
+    samDAO.userHasAction(SamResourceTypeNames.managedGroup, "Alpha_Spend_Report_Users", SamResourceAction("use"), userInfo).flatMap {
+      case true => op
+      case false => Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.Forbidden, "This API is not live yet.")))
+    }
+  }
+
+  def extractSpendReportingResults(rows: List[TableRow], startTime: DateTime, endTime: DateTime): SpendReportingResults = {
+    val currency = Currency.getInstance(rows.head.getF.get(2).getV.toString)
+
+    val dailySpend = rows.map { row =>
+      val rowCost = BigDecimal(row.getF.get(0).getV.toString).setScale(currency.getDefaultFractionDigits, RoundingMode.HALF_EVEN)
+      val rowCredits = BigDecimal(row.getF.get(1).getV.toString).setScale(currency.getDefaultFractionDigits, RoundingMode.HALF_EVEN)
+      SpendReportingForDateRange(rowCost.toString(),
+        rowCredits.toString(),
+        currency.getCurrencyCode,
+        DateTime.parse(row.getF.get(3).getV.toString),
+        DateTime.parse(row.getF.get(3).getV.toString).plusDays(1).minusSeconds(1))
+    }
+    val dailySpendAggregation = SpendReportingAggregation(
+      SpendReportingAggregationKey("total"), dailySpend
+    )
+
+    val costRollup = rows.map { row =>
+      BigDecimal(row.getF.get(0).getV.toString)
+    }.sum.setScale(currency.getDefaultFractionDigits, RoundingMode.HALF_EVEN)
+    val creditsRollup = rows.map { row =>
+      BigDecimal(row.getF.get(1).getV.toString)
+    }.sum.setScale(currency.getDefaultFractionDigits, RoundingMode.HALF_EVEN)
+
+    val spendSummary = SpendReportingForDateRange(
+      costRollup.toString(),
+      creditsRollup.toString(),
+      currency.getCurrencyCode,
+      startTime,
+      endTime
+    )
+
+    SpendReportingResults(Seq(dailySpendAggregation), spendSummary)
+  }
+
+  private def dateTimeToISODateString(dt: DateTime): String = dt.toString(ISODateTimeFormat.date())
+
+  private def getSpendExportConfiguration(billingProjectName: RawlsBillingProjectName): Future[BillingProjectSpendExport] = {
+    dataSource.inTransaction { dataAccess =>
+       dataAccess.rawlsBillingProjectQuery.load(billingProjectName)
+    }.map {
+      case Some(RawlsBillingProject(_, _, Some(billingAccount), _, _, _, _, _, Some(spendReportDataset), Some(spendReportTable), Some(spendReportDatasetGoogleProject))) =>
+        BillingProjectSpendExport(billingProjectName, billingAccount, Option(s"${spendReportDatasetGoogleProject.value}.${spendReportDataset.value}.${spendReportTable.value}"))
```

Comment:
🤔 This readability on this pattern match is tough. Since all we care about is the spendReportDataset, spendReportTable, billingProjectName and billingAccount I'm wondering if we should create a dedicated case class + `rawlsBillingProjectQuery` method to extract just that information. Happy to say we punt that to a future PR as well.

Reply:
Yeah, I think that'd be good. The other annoying part of this pattern match is that any time the `RawlsBillingProject` gets a new field we need to add a new `_`. I already refactored one of these in this file so I'll go ahead and take this out as well.